addons/mail/static/src/web/chat_window/chat_window.js

```diff
@@ -58,7 +61,19 @@ export class ChatWindow extends Component {
         });
         this.action = useService("action");
         this.contentRef = useRef("content");
-        useChildSubEnv({ inChatWindow: true });
+        useChildSubEnv({
+            inChatWindow: true,
+            messageHighlight: this.messageHighlight,
+            openPinMenu: () => {
+                this.state.activeMode = "pinned-messages";
+            },
+            closePinMenu: () => {
+                if (this.state.activeMode !== "pinned-messages") {
+                    return;
+                }
+                this.state.activeMode = "";
+            },
```

Comment:
Same remark as with discuss: could be `pinMenu: { open() {}, close() }`.

We better see the feature are available in contextual environment.